Clarify how to install pnpm (#10272)
PostHog/posthog.com: contents/handbook/engineering/developing-locally.md

2. Install the latest Node.js 18 (the version used by PostHog in production) with `nvm install 18`. You can start using it in the current shell with `nvm use 18`.

> 3. Install pnpm by running `corepack enable` and then running `corepack prepare pnpm@latest --activate`. Check it with `pnpm --version`.

Reviewer: super nit: Perhaps "validate the installation with" in place of "check it with"?

Author: ```suggestion
3. Install pnpm by running `corepack enable` and then running `corepack prepare pnpm@8.10.5 --activate`. Validate the installation with `pnpm --version`.
```

> 3. Install pnpm by running `corepack enable` and then running `corepack prepare pnpm@latest --activate`. Check it with `pnpm --version`.

Reviewer: I actually just randomly discovered that `pnpm@latest` installs v9, which on my machine will attempt to migrate our lockfiles to version 9's format. So this should actually be `pnpm@8.10.5`, which is what's mentioned in our monorepo's package.json: https://github.com/PostHog/posthog/blob/master/package.json#L18
